Parser.php

```diff
@@ -82,8 +82,19 @@ class Parser
     public function __construct(PartFactory $partFactory)
     {
         $this->partFactory = $partFactory;
+        $this->reset();
```

Comment:
I 100% get why that's a bad solution, I knew it was a bad solution when I wrote it. I am honestly concerned about re-engineering the parser. It was designed and built as a mutable object, and all of it's tests are based off of that. I don't want to potentially introduce many more bugs into such a heavily used feature.